platforms/chibios/drivers/spi_master.c

```diff
@@ -259,6 +306,10 @@ bool spi_start(pin_t slavePin, bool lsbFirst, uint8_t mode, uint16_t divisor) {
 }
 
 spi_status_t spi_write(uint8_t data) {
+#if defined(QMK_MCU_SERIES_STM32H7XX)
+    SCB_CleanDCache_by_Addr((uint32_t *)(((uint32_t)&data) & ~(uint32_t)0x1F), 1 + 32);
+#endif
```

Comment:
Apparently ChibiOS has its own implementation for the cache flushing and invalidation:
- `cacheBufferFlush(saddr, n)` — should be equivalent to `SCB_CleanInvalidateDCache_by_Addr(saddr, n)`;
- `cacheBufferInvalidate(saddr, n)` — should be equivalent to `SCB_InvalidateDCache_by_Addr(saddr, n)`.

For some reason ChibiOS does not have an equivalent for `SCB_CleanDCache_by_Addr()`, which would normally be enough for a transmit buffer which is not expected to be modified by DMA; doing `SCB_CleanInvalidateDCache_by_Addr()` instead of that would give correct results, but could also reduce performance if the data in the buffer (or just sharing a cache line with the buffer) subsequently gets read by the CPU.

Should we use the CMSIS functions or the ChibiOS macros here?  ChibiOS macros apparently exist everywhere (they are defined as doing nothing on MCUs which don't have the data cache), so may be used without ifdefs; CMSIS functions exist only when the selected MCU has the Cortex-M7 core (there are no stubs for other cores), so they would need some ifdefs.

Also the `QMK_MCU_SERIES_STM32H7XX` define is not the proper one — STM32F7xx MCUs would need the same treatment.  One possible option is to use `__DCACHE_PRESENT` (it is defined to 1 in the ST headers for their M7-based chips, and both CMSIS and ChibiOS use that macro in the implementation of the cache flushing functions.